lib/harvest_utils.rb

```diff
@@ -554,116 +559,125 @@ def self.remove_selective(provider, reindex_by)
-      record.metadata.to_s.include? @noharvest_stopword
+  def self.quarantine_and_report(record)
+    destination = "#{@quarantined_path}/#{Time.now.to_i}_#{File.basename(record)}"
+    FileUtils.mv(record, destination)
+    File.open(@log_file, "a+") do |f|
+      f << I18n.t('oai_seed_logs.text_buffer') << I18n.t('oai_seed_logs.problem_record_detected') << " #{destination}" << I18n.t('oai_seed_logs.text_buffer')
     end
+  end
 
-    def self.build_identifier(obj, provider)
-      if (provider.common_repository_type == "Islandora")
-        assembled_identifier = ''
-        url = URI.parse(obj.endpoint_url)
-        obj.identifier.select{|id| !id.include?(' ')}.each do |ident|
-          assembled_identifier = "#{url.scheme}://#{url.host}/islandora/object/#{ident}"
-        end
-        assembled_identifier
-      else
-        token = obj.send(provider.identifier_token).find {|i| i.exclude?("http")}
-        assembled_identifier = provider.identifier_pattern.gsub("$1", token)
+  def self.has_rights?(record)
+    !!(record.metadata.to_s =~ /<dc:rights(.*?)<\/dc:rights>|<dcterms:license(.*?)<\/dcterms:license>|<dcterms:RightsStatement(.*?)<\/dcterms:RightsStatement>/)
+  end
+
+  def self.has_noharvest_stopword?(record)
+    record.metadata.to_s.include? @noharvest_stopword
+  end
+
+  def self.build_identifier(obj, provider)
+    if (provider.common_repository_type == "Islandora")
+      assembled_identifier = ''
+      url = URI.parse(obj.endpoint_url)
+      obj.identifier.select{|id| !id.include?(' ')}.each do |ident|
+        assembled_identifier = "#{url.scheme}://#{url.host}/islandora/object/#{ident}"
       end
-      obj.add_identifier(assembled_identifier)
+      assembled_identifier
+    else
+      token = obj.send(provider.identifier_token).find {|i| i.exclude?("http")}
+      assembled_identifier = provider.identifier_pattern.gsub("$1", token)
     end
+    obj.add_identifier(assembled_identifier)
+  end
+
+  def self.remove_unwanted_identifiers(obj, provider)
+    obj.remove_identifier_containing(@passthrough_url) if provider.common_repository_type == 'Passthrough Workflow'
+    obj.remove_identifier_containing('viewcontent.cgi?') if provider.common_repository_type == 'Bepress'
+    obj.remove_identifier_containing('/videos/') if provider.common_repository_type == 'Bepress'
+    obj.remove_identifier_containing(' ')
+  end
 
-    def self.remove_unwanted_identifiers(obj, provider)
-      obj.remove_identifier_containing(@passthrough_url) if provider.common_repository_type == 'Passthrough Workflow'
-      obj.remove_identifier_containing('viewcontent.cgi?') if provider.common_repository_type == 'Bepress'
-      obj.remove_identifier_containing('/videos/') if provider.common_repository_type == 'Bepress'
-      obj.remove_identifier_containing(' ')
+  def has_required_fields(file)
+    valid = true
+    %w[dc:title dc:rights dc:identifier].each do |field|
+      valid = File.readlines(file).any?{|x| x.include?(field)}
```

Comment:
If the `dc:title` iteration returns `false`, and `dc:rights` iteration returns `false`, but the `dc:identifier` iteration returns `true`, then this returns `true`.

Seems like
```ruby
%w[dc:title dc:rights dc:identifier].all? do |field|
      File.readlines(file).any?{|x| x.include?(field)}
end
```
would ensure all fields are present, right?